lightning/src/ln/channel.rs

```diff
@@ -259,6 +259,9 @@ enum HTLCUpdateAwaitingACK {
 		htlc_id: u64,
 		err_packet: msgs::OnionErrorPacket,
 	},
+	FailBlindedHTLC {
```

Comment:
Yeah, good point. If that's the ultimate goal though, I think we should write these failed HTLCs as a new TLV in `Channel` rather than always failing with an empty error packet per your other ser comment.

Reply:
Ah yes, we should also write the empty error packet 👍 

Been working on this and adding the TLV on top adds a bit of complexity, unfortunately :( It feels like the right thing to do though.